curta/src/chip/uint/register.rs

```diff
@@ -70,11 +68,16 @@ impl<const N: usize> Register for ByteArrayRegister<N> {
 }
 
 impl MemoryValue for U32Register {
+    fn num_challenges() -> usize {
```

Comment:
From what I understand, the num_challenges should be equal to the number of underlying registers.  In that case, shouldn't this be 4?

Reply:
Actually it's zero because it's challenges for compressing the value to a cubic register (or 3 field elements). A U64 does byte packing so that everything actually fits into 3 field elements without needing a challenge. 

Do you think this name is confusing?

Reply:
Basically, num_challenges() indicated extra challenges the value might need for compression.

Reply:
I see.  That makes sense.